src/mbgl/gl/location_component_layer_impl.cpp

```diff
@@ -0,0 +1,21 @@
+#include <mbgl/gl/location_component_layer_impl.hpp>
+#include <mbgl/gl/location_component_layer.hpp>
+
+namespace mbgl {
+namespace style {
+
+LocationComponentLayer::Impl::Impl(const std::string& id_,
+                        LocationComponentLayerHost *host_)
+    : Layer::Impl(id_, std::string()) {
+    host = std::shared_ptr<LocationComponentLayerHost>(host_);
```

Comment:
Why is this `host` needed?

Reply:
I kept for practical reason. It was a host, initially, and it seemed simpler to leave it in its own class, and just add boilerplate around.